inventory/main/src/main/java/org/opennms/horizon/inventory/service/TagService.java

```diff
@@ -583,4 +583,17 @@ private boolean policyExists(long policyId, String tenantId) {
             return false;
         }
     }
+
+    public List<Integer> getMonitoringPoliciesByNodeId(String tenantId, long nodeId) {
+        if (nodeRepository.findByIdAndTenantId(nodeId, tenantId).isEmpty()) {
+            throw new InventoryRuntimeException("Node not found for id: " + nodeId);
+        }
+        return repository.findByTenantIdAndNodeId(tenantId, nodeId).stream()
+                .flatMap(mp -> mp.getMonitorPolicyIds().stream())
```

Comment:
nit :  for easy reading of code
```suggestion
                .flatMap(tag -> tag.getMonitorPolicyIds().stream())
```

Reply:
Done